envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -229,14 +229,15 @@ internal class EnvoySnapshotFactory(
 
         val endpoints = getServicesEndpointsForGroup(globalSnapshot, egressRouteSpecification)
 
-        val version = snapshotsVersions.version(group, clusters, endpoints)
+        val version = snapshotsVersions.version(group, clusters, endpoints, listeners)
 
         return createSnapshot(
             clusters = clusters,
             clustersVersion = version.clusters,
             endpoints = endpoints,
             endpointsVersions = version.endpoints,
             listeners = listeners,
+            listenersVersion = version.listeners,
             // for now we assume that listeners don't change during lifecycle
```

Comment:
I think we no longer assume that listeners don't change during lifecycle? So we need to update routes version logic. Routes version MUST change if listeners version changes. We would end up with "warming listeners" if we don't do it. 

Also we will be vulnerable now to "warming listeners" problem just like we were to "warming clusters". @lukidzi fix for warming clusters would be needed for listeners also: https://github.com/envoyproxy/java-control-plane/pull/131#discussion_r393350488 . For this I would at least create a task and link it here in TODO comment.

https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#resource-warming